src/main/java/com/android/volley/CacheDispatcher.java

```diff
@@ -159,6 +159,15 @@ void processRequest(final Request<?> request) throws InterruptedException {
                             new NetworkResponse(entry.data, entry.responseHeaders));
             request.addMarker("cache-hit-parsed");
 
+            if (!response.isSuccess()) {
```

Comment:
parseNetworkResponse is documented as being allowed to return null in case of an error, which means this would crash if that happens.

That being said, it seems likely that it would already crash in that situation:

- If entry.refreshNeeded(), then it will NPE on line 181 (response.intermediate=true).
- Otherwise, we call mDelivery.postResponse(). From there it depends on the ResponseDelivery implementation, but the default ExecutorDelivery seems like it would crash checking mResponse.isSuccess().

So maybe it's worth filing a separate bug to investigate whether we handle that scenario correctly or should just update the Javadoc to require that callers use Response.error rather than returning null?